client/post-editor/editor-gutenberg-opt-in-dialog/index.jsx

```diff
@@ -42,10 +45,16 @@ class EditorGutenbergOptInDialog extends Component {
 		optIn: PropTypes.func,
 		useClassic: PropTypes.func,
 		siteId: PropTypes.number,
+		wpAdminRedirectionUrl: PropTypes.string,
 	};
 
 	onCloseDialog = () => {
-		this.props.hideDialog();
+		const { isPrivateAtomic, wpAdminRedirectionUrl } = this.props;
+		if ( isPrivateAtomic ) {
+			window.location.href = wpAdminRedirectionUrl;
```

Comment:
Is there a scenario where this path could execute even if we're trying to use Gutenberg?

Reply:
I just renamed things a little bit to make it more obvious what's going on